app/grandchallenge/reader_studies/models.py

```diff
@@ -1268,6 +1296,56 @@ def _clean_widget(self):
                         f"you need to provide a default answer."
                     )
 
+    def _clean_widget_options(self):
+        is_step_size_set = self.answer_step_size is not None
+        is_min_value_set = self.answer_min_value is not None
+        is_max_value_set = self.answer_max_value is not None
+        is_number_validation_set = any(
+            [is_step_size_set, is_min_value_set, is_max_value_set]
+        )
+
+        if (
+            is_number_validation_set
+            and not self.answer_type == AnswerType.NUMBER
+            and not self.widget == QuestionWidgetKindChoices.NUMBER_INPUT
+        ):
+            # Server side number validation can only be done with AnswerType.NUMBER.
+            # Currently, client side number validation is only done with
+            # QuestionWidgetKindChoices.NUMBER_INPUT. If we allowed number
+            # validation with other widgets here the readers may not get
+            # feedback from the viewer about why their answers are rejected
+            raise ValidationError(
+                "Min and max values and the step size for answers "
+                "can only be defined in combination with the "
+                "Number Input widget for answers of type Number."
+            )
+
+        if is_step_size_set:
+            if not (is_min_value_set or is_max_value_set):
+                raise ValidationError(
+                    "When defining the answer step size, you need to also define "
+                    "the min or max value for the answer."
+                )
+
+            if is_min_value_set and is_max_value_set:
+                try:
+                    range = self.answer_max_value - self.answer_min_value
+                    StepValueValidator(self.answer_step_size)(range)
+                except ValidationError:
+                    raise ValidationError(
+                        "The difference between min and max answer value is not "
+                        "a multiple of the step size."
+                    )
```

Comment:
I've tried to work on the readability of this method. I do not see why this step size validation block is needed. Using:

```html
<form>
  <input type="number" step="0.3">
</form>
```

or

```html
<form>
  <input type="number" step="0.3" min="-1" max="1">
</form>
```

are both perfectly valid. We should keep to the HTML specifications.